woocommerce-services.php

```diff
@@ -769,10 +771,116 @@ public function show_tos_notice() {
+				return false;
+			}
+
+			$command = $_GET['wc-connect-deprecate'];
+
+			if ( 'reinstall' === $command ) {
+				$this->reinstall_plugin();
+			}
+
+			if( 'dismiss' === $command ) {
+				update_option( 'wc_connect_tos_accepted', true );
+			}
+
+			return true;
+		}
+
+		public function show_reinstall_notice() {
+			$accept_url = add_query_arg( array( 'wc-connect-deprecate' => 'reinstall' ) );
+			$decline_url = admin_url( 'plugins.php?wc-connect-deprecate=dismiss' );
+
+			?>
+			<div class="notice wcc-admin-notice">
+				<h1><?php _e( 'WooCommerce Services Plugin has Moved' ) ?></h1>
+				<a href="<?php echo esc_url( $decline_url ); ?>" style="text-decoration: none;" class="notice-dismiss" title="<?php esc_attr_e( 'Dismiss and deactivate the plugin', 'woocommerce-services' ); ?>"></a>
+				<p>
+					<b>The repository from which this plugin has been installed is deprecated. But don't worry, updating is painless!</b>
+				</p>
+				<p>
+					<?php echo __( 'Click the "Reinstall" button below to automatically delete this outdated plugin and install the new one. None of your settings will be lost.', 'woocommerce-services' ); ?>
+				</p>
+				<p>
+					<a href="<?php echo( esc_url( $accept_url ) ) ?>" class="button-primary"><?php _e( 'Reinstall' ) ?></a>
+				</p>
+			</div>
+			<?php
+		}
+
+		private function reinstall_plugin() {
+			if ( ! current_user_can( 'install_plugins' ) ) {
+				add_action( 'admin_notice', array( $this, 'reinstall_failed' ) );
+				return;
+			}
+
+			include_once( ABSPATH . '/wp-admin/includes/admin.php' );
+			include_once( ABSPATH . '/wp-admin/includes/plugin-install.php' );
+			include_once( ABSPATH . '/wp-admin/includes/plugin.php' );
+			include_once( ABSPATH . '/wp-admin/includes/class-wp-upgrader.php' );
+			include_once( ABSPATH . '/wp-admin/includes/class-plugin-upgrader.php' );
+
+			deactivate_plugins( plugin_basename( __FILE__ ) );
+			delete_plugins( array( plugin_basename( __FILE__ ) ) );
+
+			$api = plugins_api( 'plugin_information', array( 'slug' => 'woocommerce-services' ) );
+			if ( is_wp_error( $api ) ) {
+				add_action( 'admin_notice', array( $this, 'reinstall_failed' ) );
+				return;
+			}
+
+			$upgrader = new Plugin_Upgrader( new Automatic_Upgrader_Skin() );
+			$result   = $upgrader->install( $api->download_link );
+			if ( true !== $result ) {
+				add_action( 'admin_notice', array( $this, 'reinstall_failed' ) );
+				return;
+			}
+
+			$success = is_null( activate_plugin( 'woocommerce-services/woocommerce-services.php' ) );
+
+			if ( ! $success ) {
+				add_action( 'admin_notice', array( $this, 'reinstall_failed' ) );
+				return;
+			}
+
+			add_action( 'admin_notice', array( $this, 'reinstall_success' ) );
+		}
+
+		public function reinstall_failed() {
+			?>
+			<div class="notice notice-error is-dismissible">
+				<p><?php _e( 'Failed to reinstall WooCommerce Services. You may need to manually reinstall the plugin.', 'woocommerce-services' ); ?></p>
```

Comment:
At the very least, show here a link to the Plugin Install page. Ideally, redirect the user to that page directly (with this notice showing).